packages/web/src/pages/chat-page/components/ChatMessageList.tsx

```diff
@@ -54,18 +60,28 @@ const isScrolledToTop = (element: HTMLElement) => {
 }
 
 /**
- * Checks if the current message is the first unread message in the given chat
- * Used to render the new messages separator indicator
+ * Checks if the current message:
+ * - Is the first unread message
+ * - Is by a different user than the current one
  */
-const isFirstUnread = (
-  chat: UserChat,
-  message: ChatMessage,
-  currentUserId: string | null,
-  prevMessage?: ChatMessage
-) =>
-  message.created_at > chat.last_read_at &&
-  (!prevMessage || prevMessage.created_at <= chat.last_read_at) &&
-  message.sender_user_id !== currentUserId
+const shouldRenderUnreadIndicator = (
```

Comment:
want to make sure we're being thoughtful about sharing between mobile + web here (cc @dharit-tan to keep in mind) - should this live in common? a lot of this logic is really detailed and we should make sure implementations don't diverge

Reply:
yes - i made it return a boolean vs conditionally rendering the separator for this very reason. I think we can move it later when it's consumed in mobile, though